presto-product-tests/src/main/java/io/prestosql/tests/hive/ACIDTestHelper.java

```diff
@@ -0,0 +1,101 @@
+
+import com.google.common.net.HostAndPort;
+import io.prestosql.plugin.hive.metastore.thrift.NoHiveMetastoreAuthentication;
+import io.prestosql.plugin.hive.metastore.thrift.ThriftHiveMetastoreClient;
+import io.prestosql.plugin.hive.metastore.thrift.Transport;
+import io.prestosql.tempto.context.ThreadLocalTestContextHolder;
+import io.prestosql.tempto.query.QueryExecutor;
+import org.apache.hadoop.hive.metastore.api.AllocateTableWriteIdsRequest;
+import org.apache.hadoop.hive.metastore.api.Table;
+import org.apache.hadoop.hive.ql.io.AcidUtils;
+import org.apache.thrift.TException;
+
+import java.net.URI;
+import java.util.Collections;
+import java.util.Optional;
+import java.util.UUID;
+
+import static io.prestosql.tempto.assertions.QueryAssert.Row.row;
+import static io.prestosql.tempto.assertions.QueryAssert.assertThat;
+
+public final class ACIDTestHelper
+{
+    private static final String LOCALHOST = "localhost";
+
+    private ACIDTestHelper()
+    {
+    }
+
+    // This simulates a aborted transaction which leaves behind a file in a partition of a table as follows:
+    // 1. Open Txn
+    // 2. Rollback Txn
+    // 3. Create a new table with location as the given partition location + delta directory for the rolled back transaction
+    // 4. Insert something to it which will create a file for this table
+    public static void simulateAbortedHiveTransaction(String database, String tableName, String partitionSpec)
+            throws TException
+    {
+        ThriftHiveMetastoreClient client = createMetastoreClient();
+        try {
+            // Step 1
+            long transaction = client.openTransaction("test");
+
+            AllocateTableWriteIdsRequest allocateTableWriteIdsRequest = new AllocateTableWriteIdsRequest(database, tableName);
+            allocateTableWriteIdsRequest.setTxnIds(Collections.singletonList(transaction));
+            long writeId = client.allocateTableWriteIdsBatchIntr(allocateTableWriteIdsRequest).get(0).getWriteId();
+
+            // Step 2
+            client.rollbackTransaction(transaction);
+
+            // Step 3
+            Table table = client.getTableWithCapabilities(database, tableName);
+            String tableLocation = table.getSd().getLocation();
+            String partitionLocation = tableLocation.endsWith("/") ? tableLocation + partitionSpec : tableLocation + "/" + partitionSpec;
+            String deltaLocation = partitionLocation + "/" + AcidUtils.deltaSubdir(writeId, writeId, 0);
+            QueryExecutor hiveQueryExecutor = ThreadLocalTestContextHolder.testContext().getDependency(QueryExecutor.class, "hive");
+
+            String tmpTableName = getNewTableName();
+            hiveQueryExecutor.executeQuery(String.format("CREATE EXTERNAL TABLE %s (col string) stored as ORC location '%s'", tmpTableName, deltaLocation));
+
+            // Step 4
+            hiveQueryExecutor.executeQuery(String.format("INSERT INTO TABLE %s SELECT 'a'", tmpTableName));
```

Comment:
I do not like the assumption here that the layout of files generated by INSERT matches what Hive expects for a delta directory.
To validate approach  can we create another delta directory but for non-aborted transaction. And assert that rows from that directory are visible on query?

Reply:
If I understood correctly, I have made changes to insert delta after this step to validate.

Reply:
This is not exactly my point. What I would like to have, is to add data for non-rolled-backed transaction using same approach you use for rolled-back transaction. 
With creating temporary external table pointing to delta directory, and doing insert to temporary table. 

I want to have it to validate that way we use to insert data works as expected.
Without the validation there is a chance that we are not seeing rows inserted to delta-directory for rolled-back transaction, not because the transaction is rolled back, but because test is broken :)

Is it more clear?

Reply:
I like the approach, but creating an external table on a valid delta location, and then inserting data in ACID format with keeping the file name in format bucket_000XYZ, is not possible in product tests IMHO. Please let me know if you know a way around this.

To validate the 2nd point, I deleted the data present at a valid delta location. I created an external table on a valid delta location, inserted the data form this external table. But, then Presto select query failed with error: 

```
o.prestosql.spi.PrestoException: ORC ACID file should have 6 columns: hdfs://hadoop-master:9000/user/hive/warehouse/dummy_id_data/delta_0000001_0000001_0000/000000_0

at io.prestosql.plugin.hive.orc.OrcPageSourceFactory.verifyAcidSchema(OrcPageSourceFactory.java:423)
```

So, this is validating our assumption that rolled back transaction won't be read from Presto, if it is being read, then the query will fail with the error mentioned above.